src/theory/sets/theory_sets_type_rules.h

```diff
@@ -243,6 +243,31 @@ struct ChooseTypeRule
   }
 }; /* struct ChooseTypeRule */
 
+struct IsSingletonTypeRule
+{
+  inline static TypeNode computeType(NodeManager* nodeManager,
+                                     TNode n,
+                                     bool check)
+  {
+    Assert(n.getKind() == kind::IS_SINGLETON);
+    TypeNode setType = n[0].getType(check);
+    if (check)
+    {
+      if (!setType.isSet())
+      {
+        throw TypeCheckingExceptionPrivate(
+            n, "IS_SINGLETON operator expects a set, a non-set is found");
+      }
+    }
+    return nodeManager->booleanType();
+  }
+  inline static bool computeIsConst(NodeManager* nodeManager, TNode n)
+  {
+    Assert(n.getKind() == kind::IS_SINGLETON);
+    return false; // n[0].isConst();
```

Comment:
```suggestion
    return false;
```

Reply:
I why does not `n[0].const()` work here? `is_singleton (singleton 1)` is a constant. An assertion error was thrown when I used `n[0].const()`.

Reply:
Actually you should just delete this entire method.  Applications of is_singleton are never constant.

This method returns true if the node n is a canonical representation of its type. This is clearly not the case for is_singleton: it is not the canonical representation of a Boolean.